lib/Doctrine/ORM/Tools/Pagination/LimitSubqueryOutputWalker.php

```diff
@@ -92,6 +108,16 @@ public function __construct($query, $parserResult, array $queryComponents)
      */
     public function walkSelectStatement(SelectStatement $AST)
     {
+        // In the case of ordering a query by columns from joined tables, we
+        // must add those columns to the select clause of the query BEFORE
+        // the SQL is generated.
+        $this->addMissingItemsFromOrderByToSelect($AST);
+
+        // Remove order by clause from the inner query
+        // It will be re-appended in the outer select generated by this method
+        $orderByClause = $AST->orderByClause;
+        $AST->orderByClause = null;
```

Comment:
I'm not quite sure if this does any harm to non SQL Server platforms which actually support `ORDER BY` clauses in subqueries. This might change results, no? So an option could be to throw an exception for SQL Server or just ignore the `ORDER BY` as done here. Please correct me if I'm wrong here...